src/Storages/ObjectStorage/StorageObjectStorage.cpp

```diff
@@ -42,13 +42,86 @@ namespace ErrorCodes
     extern const int DATABASE_ACCESS_DENIED;
     extern const int NOT_IMPLEMENTED;
     extern const int LOGICAL_ERROR;
+    extern const int BAD_ARGUMENTS;
 }
 
 namespace StorageObjectStorageSetting
 {
 extern const StorageObjectStorageSettingsBool allow_dynamic_metadata_for_data_lakes;
 }
 
+namespace
+{
+    void sanityCheckPartitioningConfiguration(
+        const ASTPtr & table_level_partition_by,
+        const ASTPtr & query_partition_by,
+        const StorageObjectStorage::ConfigurationPtr & configuration)
+    {
+        if (!table_level_partition_by && !query_partition_by)
+        {
+            // do we want to assert that `partitioning_style` is not set to something different style AND
+            // wildcard is not set either?
+            return;
+        }
+
+        if (table_level_partition_by && query_partition_by)
+        {
+            // should never happen because parser should not allow that
+            throw Exception(ErrorCodes::LOGICAL_ERROR, "Table level partition expression and query level partition expression can't be specified together, this is a bug");
```

Comment:
Is that a users error or is it an indicator that ch code did something wrong? In former case it is better to use `BAD_ARGUMENTS` or some other error code instead of `LOGICAL_ERROR`. Since logical error is reserved for validating contracts that were failed by other pieces of code (i.e. programmer's error)

Reply:
The user can't force this occasion, there is no code path that leads to this. 

A developer could call this function providing both, and then this exception would be thrown. There is no scenario in which this would happen nowadays, tho